task/03-date-tasks.js

```diff
@@ -75,8 +79,17 @@ function isLeapYear(date) {
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,10,0,0,250)     => "00:00:00.250"
  *    Date(2000,1,1,10,0,0),  Date(2000,1,1,15,20,10,453)   => "05:20:10.453"
  */
-function timeSpanToString(startDate, endDate) {
-   throw new Error('Not implemented');
+function timeSpanToString(startDate, endDate) {  
+    let ms = (endDate-startDate)/1000;
+    let ss = Math.floor(ms%60);
+    let mm = Math.floor((ms/60)%60);
+    let hh = Math.floor((ms/3600)%60);
+    let sss = endDate.getMilliseconds() - startDate.getMilliseconds(); 
+    hh<10 ? hh='0'+hh : hh;
+    mm<10 ? mm='0'+mm : mm;
+    ss<10 ? ss='0'+ss : ss;
+    sss<10 ? sss='00'+sss : sss;
+    return hh+':'+mm+':'+ss+'.'+sss;
 }
```

Comment:
endDate-startDate is the number of milliseconds

Reply:
Now the main issue here is that you didn't have to calculate any of this values. You could use Date class to get all this information. There is a constructor in Date class that accepts the number of millisecond as an argument (https://en.wikipedia.org/wiki/Unix_time). Then there are methods like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getUTCMinutes

Reply:
or you could even use toISOString method